pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -79,6 +79,12 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 		}
 	}
 
+	// If the replicaset is failing we assume its an error we have to surface
+	if rev.Status.IsReplicaSetFailure(&deployment.Status) {
+		rev.Status.PropagateDeploymentStatus(&deployment.Status)
+		return nil
+	}
```

Comment:
My initial thought was to have this in the `else` clause. But seeing this here I wonder if there is a case where creating the Deployment will succeed but response includes a status condition with a ReplicaFailure?

Otherwise this section might be easier to reason about if we push `if !rev.Status.IsActivationRequired()` down into 
`PropagateDeploymentStatus` and we can do the replica failure check there